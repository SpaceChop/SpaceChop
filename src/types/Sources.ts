import * as t from 'runtypes';
import Source from './Source';

const Sources = t.Array(Source);

export type Sources = t.Static<typeof Sources>;
export default Sources;
