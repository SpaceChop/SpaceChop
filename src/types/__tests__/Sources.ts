import Sources from '../Sources';

describe('validate Sources', () => {
  describe('invalid', () => {
    it('should require sources', () => {
      const sources = null;
      expect(() => Sources.check(sources)).toThrowErrorMatchingSnapshot();
    });

    it('should require valid sources array', () => {
      const sources = {};
      expect(() => Sources.check(sources)).toThrowErrorMatchingSnapshot();
    });

    it('should require valid source', () => {
      const sources = [{}];
      expect(() => Sources.check(sources)).toThrowErrorMatchingSnapshot();
    });

    it('should throw on any other type of source', () => {
      const sources = [{ bad_source: {} }];
      expect(() => Sources.check(sources)).toThrowErrorMatchingSnapshot();
    });
  });
});
