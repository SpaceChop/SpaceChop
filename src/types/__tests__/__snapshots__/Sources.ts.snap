// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Sources invalid should require sources 1`] = `"Expected array, but was object"`;

exports[`validate Sources invalid should require valid source 1`] = `"Expected { http: { root: string; }; } | { s3: { access_key_id: string; secret_access_key: string; region: string; bucket_name: string; path: string; } & { endpoint?: string; }; } | { volume: { root: string; }; }, but was object"`;

exports[`validate Sources invalid should require valid sources array 1`] = `"Expected array, but was object"`;

exports[`validate Sources invalid should throw on any other type of source 1`] = `"Expected { http: { root: string; }; } | { s3: { access_key_id: string; secret_access_key: string; region: string; bucket_name: string; path: string; } & { endpoint?: string; }; } | { volume: { root: string; }; }, but was object"`;
